packages/monitors/src/OptimisticOracleContractMonitor.js

```diff
@@ -0,0 +1,198 @@
+    this.web3 = this.optimisticOracleContractEventClient.web3;
+
+    // Previous contract state used to check for new entries between calls.
+    this.lastRequestPriceBlockNumber = 0;
+    this.lastProposePriceBlockNumber = 0;
+    this.lastDisputePriceBlockNumber = 0;
+    this.lastSettlementBlockNumber = 0;
+
+    // Bot and ecosystem accounts to monitor, overridden by monitorConfig parameter.
+    const defaultConfig = {
+      logOverrides: {
+        // Specify an override object to change default logging behaviour. Defaults to no overrides. If specified, this
+        // object is structured to contain key for the log to override and value for the logging level. EG:
+        // { newPositionCreated:'debug' } would override the default `info` behaviour for newPositionCreated.
+        value: {},
+        isValid: overrides => {
+          // Override must be one of the default logging levels: ['error','warn','info','http','verbose','debug','silly']
+          return Object.values(overrides).every(param => Object.keys(this.logger.levels).includes(param));
+        }
+      }
+    };
+
+    Object.assign(this, createObjectFromDefaultProps(monitorConfig, defaultConfig));
+
+    // Validate the contractProps object. This contains a set of important info within it so need to be sure it's structured correctly.
+    const defaultContractProps = {
+      contractProps: {
+        value: {},
+        isValid: x => {
+          // The config must contain the following keys and types:
+          return Object.keys(x).includes("networkId") && typeof x.networkId === "number";
+        }
+      }
+    };
+    Object.assign(this, createObjectFromDefaultProps({ contractProps }, defaultContractProps));
+
+    // Helper functions from web3.
+    this.toWei = this.web3.utils.toWei;
+    this.toBN = this.web3.utils.toBN;
+    this.utf8ToHex = this.web3.utils.utf8ToHex;
+  }
+
+  // Queries RequestPrice events since the latest query marked by `lastRequestPriceBlockNumber`.
+  async checkForRequests() {
+    this.logger.debug({
+      at: "OptimisticOracleContractMonitor",
+      message: "Checking for RequestPrice events",
+      lastRequestPriceBlockNumber: this.lastRequestPriceBlockNumber
+    });
+
+    let latestEvents = this.optimisticOracleContractEventClient.getAllRequestPriceEvents();
+
+    // Get events that are newer than the last block number we've seen
+    latestEvents = latestEvents.filter(event => event.blockNumber > this.lastRequestPriceBlockNumber);
+
+    for (let event of latestEvents) {
+      const mrkdwn =
+        createEtherscanLinkMarkdown(event.requester, this.contractProps.networkId) +
+        ` requested a price at the timestamp ${event.timestamp} for the identifier: ${event.identifier}. ` +
+        `The ancillary data field is ${event.ancillaryData}. ` +
+        `Collateral currency address is ${event.currency}. Reward amount is ${event.reward} and the final fee is ${event.finalFee}. ` +
```

Comment:
The reward is offered by the price requester?

Reply:
Correct